reports/diversification.py

```diff
@@ -1,27 +1,33 @@
+import datetime
+from collections import defaultdict
+import pandas
 from logger import logger
+from marketplace import marketplace
 from . import calculator
 
 
-def generate_report_per_date(overall_report):
-    diversification_report_per_date = {}
-    for date, overall_data in overall_report.items():
-        diversification_report_per_date[date] = generate_report(date, overall_data)
+def generate_report_per_date(df_investiments):
+    diversification_report_per_date = defaultdict(datetime.datetime)
```

Comment:
Why a defaultdict? I find it better that it is a normal dict with this way it is much more clear if a specific date is not available.

Reply:
I see what you say. I don't have a solid argument to support a `defaultdict`. I'm going to change, but I think that this should also be changed in the overall report because both do the same.